### PR TITLE
[ERM UI] Landing page if domain contains no links

### PR DESCRIPTION
**Pull request by gherceg**
## Summary

[JIRA Ticket](https://dimagi-dev.atlassian.net/browse/SS-167)

Previously, a domain with no links associated with it would show the blank page with a message along the lines of "There are currently no linked domains". Now that users can link domains from this page, we need to update this initial state to encourage users link the current domain to others. This adds the following:

- a new page for getting started with linked projects
- if the user is active in any domains that are already upstream domains, display those with the option of adding the current domain as a downstream domain
- give the user the option link domains downstream of the current domain
- refactor django template vars to knockout vars to enable updating the current page based on observables

Screen Shot 2021-07-14 at 11 15 33 AM

## Feature Flag

`LINKED_DOMAINS`
## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### QA Plan

Will request QA on parent branch.
### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Comment by gherceg** [comment]
There is a permissions component that I need to address, and that has impacts wider than just this change. Ideally, only an admin user in both domains can create links between the two. Right now the `get_available_domains_to_link` is used to fetch domains that could be added downstream, but does not perform any admin check. Updating this will address both the path on this ticket and on the typical Add Downstream Domain path from an already existing upstream project. Any thoughts?

**Comment by gherceg** [comment]
@proteusvacuum I merged your changes into the base branch without any issues, but this PR is in the process of switching from django template vars to knockout vars, so if you could look over [this merge commit](https://github.com/dimagi/commcare-hq/pull/30052/commits/e646d1ff0893ff05d974d2594ab3b351e1b1a102) and verify that it looks good I would appreciate it. Happy to offline about those changes as well.

**Comment by gherceg** [comment]
Removed awaiting QA label because I will first merge this into gh/improve-erm-ui once approved, and run QA on that.

File: corehq/apps/linked_domain/static/linked_domain/js/domain_links.js
Comment (biyeun) [inline corehq/apps/linked_domain/static/linked_domain/js/domain_links.js:104]: how is this `computed`? it looks like `isDownstreamDomain` is not an Observable, so its value is never expected to change. If that's the case `ko.computed` isn't necessary here.

`ko.computed` is useful when its computation references live observables, but if the values never change after initialization, it's best to just use a variable.

File: corehq/apps/linked_domain/static/linked_domain/js/domain_links.js
Comment (biyeun) [inline corehq/apps/linked_domain/static/linked_domain/js/domain_links.js:373]: `upstreamButtonClass` i think is a better name

File: corehq/apps/linked_domain/templates/linked_domain/get_started.html
Comment (biyeun) [inline corehq/apps/linked_domain/templates/linked_domain/get_started.html:20]: what's the `<br/>` here for? two things..usually try to avoid extra HTML in `{% blocktrans %}` unless absolutely needed. Secondly, could this be accomplished with a `<div class="spacer"></div>` after the `<p/>`s?

File: corehq/apps/linked_domain/templates/linked_domain/get_started.html
Comment (biyeun) [inline corehq/apps/linked_domain/templates/linked_domain/get_started.html:42]: same comment with `<br />` as above

**Review by biyeun — APPROVED** [review]
some small comments. not 100% blocking, but would def like them addressed

File: corehq/apps/linked_domain/templates/linked_domain/get_started.html
Comment (gherceg) [reply corehq/apps/linked_domain/templates/linked_domain/get_started.html]: Not sure why I included this.

File: corehq/apps/linked_domain/static/linked_domain/js/domain_links.js
Comment (gherceg) [reply corehq/apps/linked_domain/static/linked_domain/js/domain_links.js:104]: Sorry I missed this comment initially. Yeah I understand when computed is useful vs when it is unnecessary. I think this came from place of optimism in that a future change could make `isDownstreamDomain` observable, and the computed var would be ready to go. However I shouldn't add unnecessary code and I'm sure a computed var has slight overhead compared to a standard var. Made a [separate PR](https://github.com/dimagi/commcare-hq/pull/30071) to fix this since I already merged this PR.

**Pull request merged**
